### PR TITLE
feat(ecr): configure immutable tags for ecr repo

### PR DESCRIPTION
**Pull request by captainrandom**
When creating an ecr repo allow tags to be immutable by setting the ImageTagMutability field found here:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-repository.html#cfn-ecr-repository-imagetagmutability

Trying to get the minor version bump script to work at the moment.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by captainrandom** [comment]
Broader question: should the `imageTagImmutable` property be a boolean or should it be a string?

File: packages/@aws-cdk/aws-ecr/lib/repository.ts [hunk 1]
```diff
@@ -324,6 +324,13 @@ export interface RepositoryProps {
    *  @default false
    */
   readonly imageScanOnPush?: boolean;
+
+  /**
+   * Enable image tag references to be mutable
+   *
+   *  @default true
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-ecr/lib/repository.ts:331]: The default for this is actually `false` right? When you pass `undefined` to the L1, then the default value is `MUTABLE` as documented [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-repository.html#cfn-ecr-repository-imagetagmutability)

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]
The following files have unnecessary changes. Would appreciate if you could clean these up.

You can do `yarn install --frozen-lockfile` to install without upgrading versions in yarn.lock.

.github/actions/prlinter/package.json
packages/cdk/package.json
packages/decdk/test/fixture/package.json
tools/yarn-cling/test/test-fixture/package1/package.json
tools/yarn-cling/test/test-fixture/package2/package.json
yarn.lock

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]
Actually I believe that the name of this property should be `imageTagMutability` to match the underlying L1 and the type should be an Enum. Even though there are only two options this feels like the best balance between safety and verbosity. The users of the L2 don't need to know to put 'MUTABLE' or 'IMMUTABLE' they just select an option on the Enum and this ensures that in the future if new options are added (idk what those would be but whatever) we can handle those.

**Comment by ap00rv** [comment]
this is duplicate of : https://github.com/aws/aws-cdk/pull/10557/files

**Comment by MrArnoldPalmer** [comment]
Closing this since #10557 got merged.

**Pull request closed without merging**